apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.py

```diff
@@ -0,0 +1,218 @@
+import base64
+import json
+from pathlib import Path
+from typing import List
+
+import chevron
+import lxml.html
+import prairielearn as pl
+from lxml.html import HtmlElement
+from typing_extensions import assert_never
+
+ATTR_ANSWER_NAME = "answers-name"
+ATTR_WIDTH = "width"
+ATTR_HEIGHT = "height"
+ATTR_SOURCE_FILE_NAME = "source-file-name"
+ATTR_SOURCE_DIRECTORY = "directory"
+ATTR_GRADABLE = "gradable"
+
+
+SOURCE_DIRECTORY_MAP = {
+    "serverFilesCourse": "server_files_course_path",
+    "clientFilesCourse": "client_files_course_path",
+    "clientFilesQuestion": "client_files_question_path",
+    ".": "question_path",
+}
+
+# -----------------------------------------------------------------------------
+# The logic below is derived from the combinations at
+# https://github.com/PrairieLearn/PrairieLearn/pull/9900#discussion_r1685163461
+
+
+def is_answer_name_required(gradable: bool) -> bool:
+    # All cases marked (ANS)
```

Comment:
```suggestion
    """All cases marked (ANS)"""
```
All top comments should be function docstrings: https://peps.python.org/pep-0257/